hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -529,6 +553,21 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		return ctrl.Result{RequeueAfter: duration}, nil
 	}
 
+	tunedConfigMap := TunedConfigMap(controlPlaneNamespace, nodePool.Name)
+	if tunedConfig == "" {
+		if err := r.Delete(ctx, tunedConfigMap); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
I think this is causing our 404 delete detection to trigger.  To avoid unneeded API server calls, we do a Get first and if that returns IsNotFound, we skip the Delete.

Reply:
Thank you Seth, good catch. I corrected this.